src/macros.rs

```diff
@@ -0,0 +1,51 @@
+//! Macros provided for convenience
+
+/// Create a heap allocator providing a heap of the given size in bytes
+///
+/// You can only have ONE allocator at most
+#[macro_export]
+macro_rules! heap_allocator {
+    ($size:expr) => {
+        #[global_allocator]
+        static ALLOCATOR: $crate::EspHeap = $crate::EspHeap::empty();
+
+        fn init_heap() {
+            static mut HEAP: core::mem::MaybeUninit<[u8; $size]> = core::mem::MaybeUninit::uninit();
+
+            unsafe {
+                ALLOCATOR.init(HEAP.as_mut_ptr() as *mut u8, $size);
+            }
+        }
+
+        init_heap();
```

Comment:
Why do we declare a function to then call it right after?

Reply:
Yes we can get rid of this - was just to resemble what we currently suggest in the examples